### PR TITLE
bug fixes Straddling checkerboard

### PR DESCRIPTION
**Pull request by t-m-z**
check tests and correct code
refacor widget => move adjustment of strings to backend
refactor logic => add function to fill key with spaces, correct typing error

### CODE REVIEW

File: lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart [hunk 1]
```diff
@@ -93,17 +106,25 @@ StraddlingCheckerboardOutput decryptStraddlingCheckerboard(
   }
 }
 
+String _fillKey(String key, bool matrix4x10) {
+  if (key.length == 10) return key;
+
+  String result = key;
+  for (int i = key.length; i < 10; i++) result = result + ' ';
```
Comment (S-Man42) [inline lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart:113]: There's a function for that: key.padRight(10)

File: lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart [hunk 2]
```diff
@@ -93,17 +106,25 @@ StraddlingCheckerboardOutput decryptStraddlingCheckerboard(
...
+  return result;
+}
+
 bool _invalidKey(String key, bool matrix4x10) {
-  int space = 0;
+  int spaces = 0;
 
   for (int i = 1; i < key.length; i++) if (key[i] != ' ') if (key.substring(0, i - 1).contains(key[i])) return true;
```
Comment (S-Man42) [inline lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart:120]: More efficient: 
key.split('').toSet().join() != key // --> toSet() removes all duplicates

This is a wonderful common util, I wonder why it does not exist yet, it is used so many times. So I recently created one:

https://github.com/S-Man42/GCWizard/commit/df8e854c10b24ffd34329c5569008e6886b97488

File: lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart [hunk 3]
```diff
@@ -93,17 +106,25 @@ StraddlingCheckerboardOutput decryptStraddlingCheckerboard(
...
 
-  for (int i = 0; i < key.length; i++) if (key[i] == ' ') space++;
+  for (int i = 0; i < key.length; i++) if (key[i] == ' ') spaces++;
```
Comment (S-Man42) [inline lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart:122]: This one as well can be done in a more Dart way:
key.replaceAll(RegExp('[^\s]'), '').length

I also added a util method:

https://github.com/S-Man42/GCWizard/commit/490d82b102bbe9d59357bbe48407111073833a49

File: test/logic/tools/crypto_and_encodings/straddling_checkerboard.dart [hunk 4]
```diff
@@ -16,15 +15,15 @@ void main() {
...
       {'input' : 'Treffen morgen Zwanzig Uhr an Treffpunkt drei', 'key': 'DEI NS TAR', 'alphabetWord' : '', 'columnOrder' : '8529406137', 'matrix4x10' : false, 'mode': PolybiosMode.AZ09, 'alphabet': '', 'expectedOutput' : '1759292546393977995463616434612996362947633463175929268624961638752'},
       {'input' : 'Treffen morgen 2000 Uhr an Treffpunkt 3', 'key': 'D   EINSTA', 'alphabetWord' : 'R', 'columnOrder' : '0123456789', 'matrix4x10' : true, 'mode': PolybiosMode.AZ09, 'alphabet': '', 'expectedOutput' : '810413134639192010144639312929293923151039963981041313212361783932'},
 
       // https://en.wikipedia.org/wiki/Straddling_checkerboard
-      {'input' : 'ATTACK AT DAWN', 'key': 'ET AON RIS', 'alphabetWord' : '', 'columnOrder' : '0123456789', 'matrix4x10' : false, 'mode': PolybiosMode.CUSTOM, 'alphabet': 'BCDFGHJKLMPQ/UVWXYZ.', 'expectedOutput' : '31132127693169223655'},
+      {'input' : 'ATTACK AT DAWN', 'key': 'ET AON RIS', 'alphabetWord' : '', 'columnOrder' : '0123456789', 'matrix4x10' : false, 'mode': PolybiosMode.CUSTOM, 'alphabet': 'BCDFGHJKLMPQUVWXYZ./', 'expectedOutput' : '31132127683168223645'},
```
Comment (S-Man42) [inline test/logic/tools/crypto_and_encodings/straddling_checkerboard.dart:26]: Can you please add a few tests for the automatic appending of the spaces please? :)

File: lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart (hunk 1 above)
Comment (t-m-z) [reply lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart:113]: Sorry - lack of knowledge. I believe there are a lot of lessons waiting for me

File: lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart (hunk 3 above)
Comment (t-m-z) [reply lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart:122]: well - wonderful idea, however I need also the original key.length - I have to use this in another way

File: lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart (hunk 2 above)
Comment (t-m-z) [reply lib/logic/tools/crypto_and_encodings/straddling_checkerboard.dart]: check - new function in common_utils found. We have to thinka about how to inform about new common utils

File: test/logic/tools/crypto_and_encodings/straddling_checkerboard.dart (hunk 4 above)
Comment (t-m-z) [reply test/logic/tools/crypto_and_encodings/straddling_checkerboard.dart:26]: done

**Pull request merged**